packages/flutter_tools/lib/src/base/build.dart

```diff
@@ -212,15 +212,19 @@ class AOTSnapshotter {
 
   /// Builds an iOS framework at [outputPath]/App.framework from the assembly
   /// source at [assemblyPath].
-  Future<RunResult> _buildIosFramework({
+  Future<RunResult> _buildFramework({
     @required IOSArch iosArch,
```

Comment:
How is this working on macOS if the arch handling is still hard-coded to iOS?

Reply:
I confusingly added an x86 iOS arch and conditionally excluded the `-miphoneos-version-min=*` flag. That was enough to get AOT working. I'll plan on opening a separate PR to add the rules for the actual AOT release/profile build